src/controllers/controller.h

```diff
@@ -193,13 +196,33 @@ class ControllerJSProxy : public QObject {
             : m_pController(m_pController) {
     }
 
+    QHash<QString, QString> m_cache;
+
     // The length parameter is here for backwards compatibility for when scripts
     // were required to specify it.
     Q_INVOKABLE virtual void send(const QList<int>& data, unsigned int length = 0) {
         Q_UNUSED(length);
         m_pController->send(data, data.length());
     }
 
+    // Available charsets should be available here:
+    // http://www.iana.org/assignments/character-sets/character-sets.xhtml
+    Q_INVOKABLE QByteArray convertEncoding(const QString& targetCharset, const QString& value) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+        auto* codec = QTextCodec::codecForName(targetCharset.toUtf8());
```

Comment:
It seems like I can't `delete codec` because `QTextCodec` destrucotr is `protected`. Won't this cause a resource leak problem?

Reply:
only if the pointer returned is owning. Thats why plain pointers are bad, they could be owning or non-owning and you never know (nor can be sure either is always the case) because that information is not in the type system. In this case, I'm fairly certain its non-owning so you don't need to do anything with the pointer afterwards (it could just as well be a reference being returned instead of a pointer). This is probably why the destructor is protected, to protect the user from calling delete on it.